tests/lightning_qubit2/test_lightning_qubit2.py

```diff
@@ -0,0 +1,588 @@
+            _ = LightningQubit2(wires=2, shots=n_shots, mcmc=True, num_burnin=num_burnin)
+
+    def test_invalid_kernel_name(self):
+        """Test that an error is raised when the kernel_name is not "Local" or "NonZeroRandom"."""
+
+        _ = LightningQubit2(wires=2, shots=1000, mcmc=True, kernel_name="Local")
+        _ = LightningQubit2(wires=2, shots=1000, mcmc=True, kernel_name="NonZeroRandom")
+
+        with pytest.raises(
+            NotImplementedError, match="only 'Local' and 'NonZeroRandom' kernels are supported"
+        ):
+            _ = LightningQubit2(wires=2, shots=1000, mcmc=True, kernel_name="bleh")
+
+
+class TestExecution:
+    """Unit tests for executing quantum tapes on LightningQubit2"""
+
+    @staticmethod
+    def calculate_reference(tape):
+        device = DefaultQubit(max_workers=1)
+        program, _ = device.preprocess()
+        tapes, transf_fn = program([tape])
+        results = device.execute(tapes)
+        return transf_fn(results)
+
+    @staticmethod
+    def process_and_execute(dev, tape):
+        program, _ = dev.preprocess()
+        tapes, transf_fn = program([tape])
+        results = dev.execute(tapes)
+        return transf_fn(results)
+
+    _default_device_options = {
+        "c_dtype": np.complex128,
+        "batch_obs": False,
+        "mcmc": False,
+        "kernel_name": None,
+        "num_burnin": None,
+    }
+
+    @pytest.mark.parametrize(
+        "config, expected_config",
+        [
+            (
+                DefaultExecutionConfig,
+                ExecutionConfig(
+                    grad_on_execution=True,
+                    use_device_gradient=False,
+                    device_options=_default_device_options,
+                ),
+            ),
+            (
+                ExecutionConfig(gradient_method="best"),
+                ExecutionConfig(
+                    gradient_method="adjoint",
+                    grad_on_execution=True,
+                    use_device_gradient=True,
+                    device_options=_default_device_options,
+                ),
+            ),
+            (
+                ExecutionConfig(
+                    device_options={
+                        "c_dtype": np.complex64,
+                        "mcmc": True,
+                    }
+                ),
+                ExecutionConfig(
+                    grad_on_execution=True,
+                    use_device_gradient=False,
+                    device_options={
+                        "c_dtype": np.complex64,
+                        "batch_obs": False,
+                        "mcmc": True,
+                        "kernel_name": None,
+                        "num_burnin": None,
+                    },
+                ),
+            ),
+            (
+                ExecutionConfig(
+                    gradient_method="backprop", use_device_gradient=False, grad_on_execution=False
```

Comment:
Why is `backprop` ok here?

Reply:
The function being tested just updates the execution config. Errors are raised when the transform program returned by `device.preprocess` is applied to the tape(s).